CMakeLists.txt

```diff
@@ -261,3 +263,95 @@ if(CELERITAS_USE_Doxygen)
 endif()
 
 #----------------------------------------------------------------------------#
+# CONFIG FILE INSTALLATION
+#----------------------------------------------------------------------------#
+
+# Where to configure cmake files
+set(CELER_CMAKE_CONFIG_DIRECTORY "${PROJECT_BINARY_DIR}/share")
+# Where to install configured cmake files
+set(CELER_INSTALL_CMAKECONFIGDIR
+  "${CMAKE_INSTALL_DATAROOTDIR}/cmake/Celeritas")
```

Comment:
As Celeritas does provide binary libraries, would `${CMAKE_INSTALL_LIBDIR}/cmake/Celeritas` be a better location? The config/targets file(s) nominally include architecture information, and there's no huge reproduction of info should multiple arches be present.

Reply:
Ah, I didn't realize that the `share|lib|lib/<arch>` options in [find_package search paths](https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure) were related to the kind of library being installed. I did look beforehand at what other projects install to and there's basically no consistency:
- `<prefix>`: ROOT
- `lib/cmake/<project>`: libxml, becgeom, openblas, gtest, veccore
- `lib/cmake/<project>-<version>`: expat, clhep, geant4
- `share/<project>`: hepmc3
- `share/cmake/<project>`: hdf5

Seems like `lib/cmake/<project>` is the winner though, so I'll change to that.

Reply:
Sounds good! Geant4'll be moving to `lib/cmake/<project>` on the next minor release.